contracts/src/token/erc721/extensions/consecutive.rs

```diff
@@ -994,5 +1034,300 @@ mod tests {
             Error::Erc721(erc721::Error::NonexistentToken(ERC721NonexistentToken { token_id }))
             if token_id == U256::from(non_consecutive_token_id)
         ));
+
+        // After being burnt the token should not be burnt again.
+        let non_existent_token = non_consecutive_token_id;
+        let err = contract
+            ._burn(non_existent_token)
+            .expect_err("should return Error::NonexistentToken");
+
+        assert!(matches!(
+            err,
+            Error::Erc721(erc721::Error::NonexistentToken (ERC721NonexistentToken{
+                token_id: t_id
+            })) if t_id == non_existent_token
+        ));
+    }
+
+    #[motsu::test]
+    fn safe_transfer_from(contract: Erc721Consecutive) {
+        let alice = msg::sender();
+        let token_id = random_token_id();
+        contract._mint(alice, token_id).expect("should mint a token to Alice");
+
+        contract
+            .safe_transfer_from(alice, BOB, token_id)
+            .expect("should transfer a token from Alice to Bob");
+
+        let owner = contract
+            .owner_of(token_id)
+            .expect("should return the owner of the token");
+
+        assert_eq!(owner, BOB);
+    }
+
+    #[motsu::test]
+    fn safe_transfers_from_approved_token(contract: Erc721Consecutive) {
+        let alice = msg::sender();
+        let token_id = random_token_id();
+        contract._mint(BOB, token_id).expect("should mint token to Bob");
+        contract.erc721._token_approvals.setter(token_id).set(alice);
+        contract
+            .safe_transfer_from(BOB, alice, token_id)
+            .expect("should transfer Bob's token to Alice");
+        let owner = contract
+            .owner_of(token_id)
+            .expect("should return the owner of the token");
+        assert_eq!(owner, alice);
+    }
+
+    #[motsu::test]
+    fn error_when_safe_transfer_transfers_from_incorrect_owner(
```

Comment:
I would skip the naming convention we decided some time ago and leave: `error_when_safe_transfers_from_incorrect_owner`